modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,154 @@
+	var/static/list/has_taur_horse_sprite = list()
+
+	/// has_taur_sprite_suit[icon_state]: has_taur_sprite but for /obj/item/clothng/suit
+	var/static/list/has_taur_sprite_suit = list()
+	/// has_taur_snake_sprite_suit[icon_state]: has_taur_snake_sprite but for /obj/item/clothing/suit
+	var/static/list/has_taur_snake_sprite_suit = list()
+	/// has_taur_paw_sprite_suit[icon_state]: has_taur_paw_sprite but for /obj/item/clothing/suit
+	var/static/list/has_taur_paw_sprite_suit = list()
+	/// has_taur_horse_sprite_suit[icon_state]: has_taur_horse_sprite but for /obj/item/clothing/suit
+	var/static/list/has_taur_horse_sprite_suit = list()
+
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+// NOTE FOR BOTH HANDLE_TAUR PROCS: These CAN be converted to handle other things, like digi sprites, or to be ran on init of ANY item. But be aware: For every dependancy you
+// add (digi sprites), you also add a for loop on init, which costs CPU time. The bigger the lists, and the more lists, the more memory we use. We want to use the least amount
+// of lists and loops as possible, and the cache system I set up doesn't completely get rid of the costs.
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Quits instantly if mutant_variants has all taur styles (STYLE_TAUR), then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing if the
+ * clothing doesnt already have it.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite for that icon state will be set to false.
+ *
+ * DO NOT RELY ON THIS PROC TO DO YOUR WORK FOR YOU. If you are adding a new item, or a sprite, you SHOULD be manually setting your mutant_variants flags, otherwise the for
+ * loop may be ran and cause slight performance loss.
+ */
+/obj/item/clothing/proc/handle_taur_sprites()
+
+	if (istype(src, /obj/item/clothing/suit))
+		handle_taur_sprites_for_suits()
+		return
+
+	if (!(istype(src, /obj/item/clothing/under))) //remove this if we ever make a taur sprite for anything else, this is for performance
+		return
+
+	if (mutant_variants & STYLE_TAUR)
+		return
+
+	if (has_taur_sprite[icon_state])
+		if (has_taur_snake_sprite[icon_state] && (!(mutant_variants & STYLE_TAUR_SNAKE)))
+			mutant_variants |= STYLE_TAUR_SNAKE
+		if (has_taur_horse_sprite[icon_state] && (!(mutant_variants & STYLE_TAUR_HOOF)))
+			mutant_variants |= STYLE_TAUR_HOOF
+		if (has_taur_paw_sprite[icon_state] && (!(mutant_variants & STYLE_TAUR_PAW)))
+			mutant_variants |= STYLE_TAUR_PAW
+		return // If we already know this icon state has a taur sprite, skip the for loop and take from the cache
```

Comment:
Setting a bitflag is just as expensive as checking to see if it's set, you don't need to check if it's already set before setting it

Reply:
But if I don't check to see if it already has the bitflag, won't it just be applied twice? Doesn't that have consequences?

Reply:
nope, they're just binary 1s in a number
the | operator says set the output to the 1s from either value. It's binary OR
so 001 |  010 becomes 011. 
the nice thing about | is it doesn't add, it's a bit operation.
So 001 | 001 is just 001, since while they share the same bit in the same position, the | operator doesn't really care